packages/block-library/src/image/index.php

```diff
@@ -24,6 +24,11 @@ function render_block_core_image( $attributes, $content ) {
 			$content = str_replace( '<img', '<img ' . $data_id_attribute . ' ', $content );
 		}
 	}
+
+	if ( isset( $attributes['border-radius'] ) ) {
+		$content = str_replace( '<figure', "<figure style='border-radius:" . $attributes['border-radius'] .  ";overflow:hidden;'", $content);
```

Comment:
despite the lines above using `str_replace` this might be a good place to use the new tag processor. might be worth modifying the whole function so that it's not split between two approaches.

of note: if the `figure` already had a `style` attribute in it then this change would wipe out that `style` attribute, making it a duplicate attribute that browsers ignore.

```php
function render_block_core_image( $attributes, $content ) {
	if ( isset( $attributes[ 'data-id'] ) ) {
		$tags = new WP_HTML_Tag_Processor( $content );
		/*
		 * The data-id="$id" attribute on the img element provides
		 * backwards-compatibility for the gallery block, which now
		 * wraps image blocks within its inner blocks.
		 *
		 * @see the `render_block_data` hook in the core/gallery block
		 */
		if ( ! $tags->next_tag( 'img' ) ) {
			break;
		}

		$tags->set_attribute( 'data-id', $attributes['data-id'] );
		$content = (string) $tags;
	}

	if ( isset( $attributes['border-radius'] ) ) {
		$tags = new WP_HTML_Tag_Processor( $content );
		if ( ! $tags->next_tag( 'figure' ) ) {
			break;
		}

		$style  = $tags->get_attribute( 'style' );
		$border = "border-radius: {$attributes['border-radius']}; overflow: hidden;";
		$tags->set_attribute( 'style', "{$style}; {$border}" );
		$content = (string) $tags;
	}

	return $content;
}
```